### PR TITLE
Add Spring Cloud Gateway

### PR DESCRIPTION
**Pull request by juliensadaoui**
related at #1500

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/gateway/infrastructure/primary/rest/GatewayResource.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  private final GatewayApplicationService gatewayApplicationService;
+
+  public GatewayResource(GatewayApplicationService gatewayApplicationService) {
+    this.gatewayApplicationService = gatewayApplicationService;
+  }
+
+  @Operation(summary = "Add Spring Cloud Gateway")
+  @ApiResponse(responseCode = "500", description = "An error occurred while adding Spring Cloud Gateway")
+  @PostMapping
+  @GeneratorStep(id = GeneratorAction.EUREKA_CLIENT)
```
Comment (Bolo89) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/springcloud/gateway/infrastructure/primary/rest/GatewayResource.java:29]: The generatorstep id should be '' springCloudGateway '' and not eurekaclient ?

File: src/main/resources/generator/server/springboot/springcloud/gateway/java/GatewayResource.java.mustache [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import reactor.core.publisher.Flux;
+import org.springframework.http.*;
+import org.springframework.web.bind.annotation.*;
+
+/**
+ * REST controller for managing Gateway configuration.
+ */
+@RestController
+@RequestMapping("/api/gateway")
+public class GatewayResource {
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/springcloud/gateway/java/GatewayResource.java.mustache:21]: No need public

File: src/main/resources/generator/server/springboot/springcloud/gateway/java/RouteVM.java.mustache [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+
+import org.springframework.cloud.client.ServiceInstance;
+
+/**
+ * View Model that stores a route managed by the Gateway.
+ */
+public class RouteVM {
+
+    private String path;
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/springcloud/gateway/java/RouteVM.java.mustache:12]: Indentation missing prettier here

File: src/main/resources/generator/server/springboot/springcloud/gateway/java/test/GatewayResourceIT.java.mustache [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.function.Predicate;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.mockito.Mockito.when;
+
+@WebFluxTest(GatewayResource.class)
+@AutoConfigureWebTestClient
+class GatewayResourceIT {
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/springcloud/gateway/java/test/GatewayResourceIT.java.mustache:38]: Need prettier to reorganize a little bit the imports

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Comment by juliensadaoui** [comment]
Bounty claimed at https://opencollective.com/generator-jhipster/expenses/76937

**Pull request merged**